### DISCUSSION TITLE
Cannot upload file changes/creation

### DISCUSSION THREAD

**Discussion by mak163** [abraunegg/onedrive#1907] [Q&A]
Every time I make changes to a file on the local disk, the service gives an error Cannot upload file changes/creation: Test/Test.txt.part: No such file or directory

**Comment by abraunegg** [comment]
@mak163 
Being a little more descriptive in your details might be a little more helpful ....

For example:
* what application version are you running
* what are your application logs detailing
* how are you editing the file 
* what is your process flow

What you have posted above is akin to claiming 'I have a flat tyre' but providing zero details to anyone to help - such as is this a bike, car, truck, bus, scooter, what were you doing when you got the flat tyre - at home, on the road, out in the bush or snow ... 

Please can you update your post with quite a few more details as its very hard to assist you with what you have currently posted

**Reply by mak163** [reply]
@abraunegg

- application version 2.4.16 from AUR https://aur.archlinux.org/packages/onedrive-abraunegg
- /var/log/onedrive/ is empty
- nano editor
- I open a file in the CherryTree note editor  that is located in the OneDrive folder. After the changes are complete, I save the file in the editor. Service give an error "Cannot upload file changes/creation: Test/Test.txt.part: No such file or directory"

**Reply by abraunegg** [reply]
@mak163 
OK .. so this will be a *very* short lived file that is created by 'nano' ... will be similar to the 'vi' and 'vim' problem of short lived files and how these short lived files, when running as a systemd service, trigger the upload of the temporary file - but, when it is attempted to be uploaded, the file is no longer there.

Your options to resolve:
* Ignore it .. it is being caused by your method of editing files. It's harmless.
* Edit your configuration so that the application ignores '.part' files and does not attempt to upload them
  * Example: `skip_file = "~*|.~*|*.tmp|*.part"`
* Use a different editor

**Reply by mak163** [reply]
@abraunegg There was already a similar line in my configuration file, but it did not affect the appearance of errors during the exchange.

**Reply by abraunegg** [reply]
@mak163 
Then your client is not using the configuration file you *think* it is using ......

**Reply by mak163** [reply]
@abraunegg That's not so. I have 6 folders in this configuration file that the client ignores when syncing.

**Reply by abraunegg** [reply]
@mak163 
> @abraunegg That's not so. I have 6 folders in this configuration file that the client ignores when syncing.

Please raise a bug following the correct process and provide verbose debug logs so that it can be identified what is going on.

99% certain right now that this *issue* is something local to you and your configuration.

**Comment by mak163** [comment]
@abraunegg It may seem strange, but after some system updates that took place today, the error stopped appearing. I will monitor the situation further.

**Comment by mak163** [comment]
@abraunegg The error is back again. The system settings have not changed.

**Comment by abraunegg** [comment]
@mak163 
Did you update your client version?

What are the messages you are getting ?

**Comment by mak163** [comment]
@abraunegg I have ArchLinux. Packages are updated automatically.
Version 2.4.16 from the AUR  is now installed

Cannot upload file changes/creation: CherryTree.ctb-journal.: No such file or directory
Cannot upload file changes/creation: CherryTree.ctb!.: No such file or directory
Cannot upload file changes/creation: CherryTree.ctb!-journal.: No such file or directory

**Comment by abraunegg** [comment]
@mak163 
Please understand the error is generated by your OS .. not this application.

Suggest that you:
* rebuild your client using the aur git package
* rebuild your client from source

The error is reported because of issues with either:
* your file system
* your physical disk
